stm32/libcanard/bxcan/src/bxcan.c

```diff
@@ -0,0 +1,753 @@
+                    if (i == 2U)
+                    {
+                        bxcan_base->TSR = BXCAN_TSR_ABRQ2;
+                    }
+                    *error_iface = true;  // TX timeout error occurred.
+                }
+
+                if (tme[i])  // This TX mailbox is free, we can use it
+                {
+                    tx_mailbox = i;
+                }
+                else  // This TX mailbox is pending, check for priority inversion
+                {
+                    if (extended_can_id >= convertRegisterToFrameID(bxcan_base->TxMailbox[i].TIR))
+                    {
+                        // There's a mailbox whose priority is higher or equal the priority of the new frame.
+                        // Priority inversion would occur! Reject transmission. We do this by insuring that
+                        // the tx_mailbox value is set to the trip value of 0xFF and break the search loop.
+                        // The priority inversion logic then catches the condition & handles it.
+                        tx_mailbox = 0xFF;
+                        break;
+                    }
+                }
+            }
+
+            if (tx_mailbox == 0xFF)
+            {
+                // All TX mailboxes are busy (this is highly unlikely); at the same time we know that there is no
+                // higher or equal priority frame that is currently pending. Therefore, priority inversion has
+                // just happend (sic!), because we can't enqueue the higher priority frame due to all TX mailboxes
+                // being busy. This scenario is extremely unlikely, because in order for it to happen, the application
+                // would need to transmit 4 (four) or more CAN frames with different CAN ID ordered from high ID to
+                // low ID nearly at the same time. For example:
+                //  1. 0x123        <-- Takes mailbox 0 (or any other)
+                //  2. 0x122        <-- Takes mailbox 2 (or any other)
+                //  3. 0x121        <-- Takes mailbox 1 (or any other)
+                //  4. 0x120        <-- INNER PRIORITY INVERSION HERE (only if all three TX mailboxes are still busy)
+                // This situation is even less likely to cause any noticeable disruptions on the CAN bus. Despite that,
+                // it is better to warn the developer about that during debugging, so we fire an assertion failure here.
+                // It is perfectly safe to remove it.
+                // Note: this error condition (all TX mailboxes busy) also occurs when erroneously sending data to the
+                // TX mailboxes of a non-existing CAN interface. Eg: accessing CAN2 on a device that only has CAN1.
+#if BXCAN_CONFIG_DEBUG_INNER_PRIORITY_INVERSION
+                BXCAN_ASSERT(!"CAN PRIO INV");  // NOLINT
+#endif
+                prio_inversion = true;
+            }
+            else
+            {
+                prio_inversion = false;
+            }
```

Comment:
I don't think you need `prio_inversion` and this condition here at all. Please move your `BXCAN_ASSERT(!"CAN PRIO INV");` into the branch `if (extended_can_id >= convertRegisterToFrameID(bxcan_base->TxMailbox[i].TIR))` on line 466.

Reply:
The condition was removed and the assert folded into the branch.